quiffen/core/account.py

```diff
@@ -123,16 +141,14 @@ def add_transaction(
         RuntimeError
             If there is no header provided, and no ``last_header`` set.
         """
-
         if not header and not self._last_header:
             raise RuntimeError('No header provided, and no last header set.')
 
         if not header:
             header = self._last_header
         else:
             try:
-                header = AccountType(header.strip().split(':')[-1])
-                self._last_header = header
+                self._last_header = self._convert_deprecated_str_header(header)
```

Comment:
Reading this again, I'm perhaps a little confused. If we were always converting to an account type, could we not just change the type hint for the `transactions` field?

Reply:
We were not necessarily always converting since you can also call `.set_header()` manually outside of this function

Reply:
True, but surely we could just do `self.header = AccountType(header)` there? That should work fine, and throw and error if it's an invalid account type, which we want? I don't mind people being able to pass in strings (hence the strings everywhere in the docs and tests), as long as things end up the right types in the objects. Also, everything is pretty well type-hinted, so people's IDEs should scream at them if they try and use the wrong things in the wrong places.

At the end of the day, there's only so much defensive programming we can do. Python is inherently flexible, and you'll always be able to break things like this.

Reply:
I'm happy to change that, but that could potentially be a breaking change, if people used `.set_account()` to set other strings that are not valid QIF account types. Shall I?

Reply:
It's not a breaking change if it fixes unintended behaviour 🤷🏻‍♂️ 

I think we should probably fix this, then run a static type checker like `pyright` against the codebase. I'm also tempted to introduce `black` for formatting. That's what I've done with [iTmpl](https://github.com/isaacharrisholt/itmpl) and it makes life so much easier.

Reply:
Ok, fine with me!

Reply:
Let's just make sure to squash the commits on this PR before merging 😅

Reply:
sure, will do :D